.github/workflows/ci-pipeline.yml

```diff
@@ -5,16 +5,50 @@ on:
     paths-ignore:
       - 'documentation/**'
       - 'README.md'
-  pull_request:
-    branches:
-      - main
+#  pull_request:
+#    branches:
+#      - main
 
 jobs:
-  build:
+  build-test:
     name: Build and test
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
       - run: docker build --target test --tag todo-app:test .
       - run: docker run --env-file .env.test todo-app:test
+ 
+  azure-deploy:
+    name: deploy to Azure
+    runs-on: ubuntu-latest
+    needs: build-test
+    if: github.event_name == 'push' && github.ref == 'main'
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERUSERNAME }}
+          password: ${{ secrets.DOCKERTOKEN }}
+
+      - name: 'Login via Azure CLI'
+        uses: azure/login@v1
+        with:
+          creds: ${{ secrets.AZURE_CREDENTIALS }}
+
+      - run: docker build --target production --tag paulcurren/todo-app:prod .
+      - run: docker push paulcurren/todo-app:prod
 
+      - name: 'Deploy to Azure Container Instances'
```

Comment:
It's pretty neat that you got your image running on Container Instances.
That said you should really be using App Service. It will become clearer why in the future modules, where container instances are inadequate for completing the exercises.

Check out the [Project Exercise ](https://project-exercises.devops.corndel.com/exercises/m8_exercise) again and follow the steps for setting up an app service hosted application and update this step to deploy there instead.